### ISSUE TITLE
Error: Error invoking remote method 'doRPC_IPC': NO_CONNECTION

### ISSUE THREAD

**Issue by Ukjenten** [ZcashFoundation/zecwallet#388]
I can't use Zecwallet anymore. 
It only says - Error: Error invoking remote method 'doRPC_IPC': NO_CONNECTION
I got this when updating from 1.8.0 i think to 1.8.3. I have been waiting for a new release and it is still the same.
![2022-10-27 19_45_59-New Issue · ZcashFoundation_zecwallet](https://user-images.githubusercontent.com/116749278/198361742-d40d56e8-6891-4768-9e0b-94d61a3c3cf2.png)

**Comment by balkanunion** [comment]
Error: Error invoking remote method 'doRPC_IPC': NO_CONNECTION

Why is this happening??

**Comment by bloxster** [comment]
I get the same error on a MacOS 13.0.1 (22A400)

**Comment by Garis2019** [comment]
Version 1.7.8 worked fine without problems. What next? All versions from 1.8.0 to 1.8.5 with RPC issue. I do anything: I changed the config many times (included nodes and various permissions and prohibitions there); in antivirus made exceptions for: Zecwallet Fullnode.exe, zcashd.exe, zcash-cli.exe, zcashd-wallet-tool.exe; prescribed rules for Internet networks (there also prescribed exceptions for ports); deleted completely (only left blocks) and manually cleaned the registry and much more, which can be discussed here for a very long time.
I think that the author (Aditya Kulkarni) probably does not understand that he created huge problems for us - his clients. We are, for the most part, just users. Why do we need to understand all this? We need to receive and transfer Zec. I have reviewed a lot of literature and help all over the Internet, but everyone just copy-pastes information from Github on their sites. Why, it is not clear (!?) And here, too, there is no clear, step-by-step instruction for troubleshooting problems with RPC anywhere.
I think it's useless to wait for help. Apparently you need to wait until the author wants to fix all the bugs and troubles, for example, in future version 1.9.0, which will be tailored for various operating systems (Win 7x64 Ultimate - Win 10, Mac and others). IMHO

**Comment by skironDotNet** [comment]
The problem is the wallet officially comes with "embed full node" what it means is when you look into 
`/opt/'Zecwallet Fullnode'/resources/bin/linux/` you can see zcashd

The only was I was able to make the wallet work is by manually executing 
`/opt/'Zecwallet Fullnode'/resources/bin/linux/./zcashd`
but then the GUI complains to run `zcashd-wallet-tool` done this, same problem.

I think ZcashFoundation and a user or repo adityapk00 is all ran by gov, to give people fake rigged wallets and tools and light wallets supporting shield addresses just to spy on user transactions. I also think US Gov is attacking blockchain to force people to use lightwalletd that is rigged server to spy.

Dear author, please assure me otherwise

**Comment by skironDotNet** [comment]
I was able to make it run, so first thing needed is the zcash conf located in ~/.zcash/zcash.conf
The minimum conf is
```
server=1
rpcuser=zecwallet
rpcpassword=somepassword
rpcallowip=127.0.0.1
```
If you want to keep blockchain on a separate drive or other location that is not default home location add 
`datadir=your-path-to-blockchain`

If you want to add custom wallet name
`wallet=hd1.wallet`

Now start "embed" zcashd node
`/opt/'Zecwallet Fullnode'/resources/bin/linux/./zcashd`

After blockchain is loaded (doesn't have to be yet fully synced) start GUI

Now the GUI will show you sync progress. I don't know what will happen when fully synced if will complain about using `zcashd-wallet-tool` also I don't know what funds are not shown but when you call `./zcash-cli getwalletinfo` it will show balance, so there is still some bug IMO

My assumption is that `Error: Error invoking remote method 'doRPC_IPC': NO_CONNECTION` is a result of GUI wallet not having permission to start `zcashd`?

**Comment by dismad** [comment]
Its not clear if you guys are using Windows, Linux, or Mac, but I recommend using zcashd + ZecwalletFullnode. Here is a quick video showing some setup issues you may encounter. I have been using Zecwallet Fullnode for a while now , so it *does work*. 

https://www.youtube.com/watch?v=9t2LX3HFldw

**Comment by skironDotNet** [comment]
@dismad the "problem" with your video is that you start zcashd manually by yourself, of course it works. What should really work is when you start GUI only, it should run zcashd in background.

Also you "cheat" the users because you said "my node is already synced". The problem is even if you start zcashd manually (or above would be solved) then GUI still complains about problem with connection when zcashd is not fully synced, while instead it should show something like "waiting for zcashd to fully sync, before showing balances"

**Comment by bolexator** [comment]
> @dismad the "problem" with your video is that you start zcashd manually by yourself, of course it works. What should really work is when you start GUI only, it should run zcashd in background.
> 
> Also you "cheat" the users because you said "my node is already synced". The problem is even if you start zcashd manually (or above would be solved) then GUI still complains about problem with connection when zcashd is not fully synced, while instead it should show something like "waiting for zcashd to fully sync, before showing balances"

You are absolutely right. The advice is complete nonsense for the most part. I have tried different ways and none of them work. It's not even about at least restore the wallet and transfer funds from there. The whole program does not work properly. It can be installed from scratch on any PC configurations and different systems - it doesn't work.

**Comment by dchmelik** [comment]
I have similar problems but trying what skironDotNet said.
